File: backend/PlzGeo.Api/Dtos/CreateVisualizationDto.cs
using PlzGeo.Api.Models;

namespace PlzGeo.Api.Dtos
{
    public abstract class CreateVisualizationDto
    {
        public string Name { get; set; } = string.Empty;

        public VisualizationType Type { get; set; }

        public List<VisualizationValueDto> Values { get; set; } = [];
    }
}

File: frontend/PlzGeo.Client/src/app/api-clients/dtos/create-visualization.dto.ts
import { VisualizationType } from '../../models/visualization-type.enum';
import { VisualizationValueDto } from './visualization-value.dto';

export interface CreateVisualizationDto {
  name: string;
  type: VisualizationType;
  values: VisualizationValueDto[];
}
